CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/utility/SaveResultsFileWriterNGTest.java

```diff
@@ -53,48 +82,152 @@ public SaveResultsFileWriterNGTest() {
+        Pattern pattern = Pattern.compile("[0-9]{8}T[0-9]{9}-ExampleClass.xml");
+        Matcher matcher = pattern.matcher(filename.toString());
+        Assert.assertTrue(matcher.matches(), "Testing generated filename matches pattern - yyyyMMddTHHmmssSSS.");
     }
-
+        
     /**
-     * Test of generateFilename method, of class SaveResultsFileWriter.
+     * Test of writeRecordStore method for case when DataAccessPreferenceKeys.getDataAccessResultsDir() returns null.
+     * Expectation is that no attempted writes are made.
      */
     @Test
-    public void testGenerateFilenameSuffix() {
+    public void testWriteRecordStoreNoDataAccessResultsDir() {
         Plugin plugin = new ExampleClass();
-        String filename = SaveResultsFileWriter.generateFilename(plugin, "xml");
-        Assert.assertTrue(filename.endsWith(".xml"));
+        TabularRecordStore tabularRecordStore = new TabularRecordStore();
+                
+        mockedDataAccessPreferenceKeys.when(() -> DataAccessPreferenceKeys.getDataAccessResultsDir()).thenReturn(null);
+        try {
+            SaveResultsFileWriter.writeRecordStore(plugin, tabularRecordStore);
+            Assert.assertTrue(true, "Testing writing record to non existant directory.");
+        } catch (Exception ex) {
+            Assert.assertTrue(false, "Exception is not expected.");
+        }
     }
+     
+    /**
+     * Test of writeRecordStore method for case when DataAccessPreferenceKeys.getDataAccessResultsDir() returns an
+     * invalid directory. Because directory (and hence generated file path) is invalid, and exception will be thrown
+     * and ConstellationLoggerHelper.exportPropertyBuilder called with appropriate ERROR status.
+     */
+    @Test
+    public void testWriteRecordStoreInvalidDataAccessResultsDir() {
+        Plugin plugin = new ExampleClass();
+        TabularRecordStore tabularRecordStore = new TabularRecordStore();
+
+        mockedDataAccessPreferenceKeys.when(() -> DataAccessPreferenceKeys.getDataAccessResultsDir()).thenReturn(new File("/BADDIR/"));
+        mockedConstellationLoggerHelper.when(() -> ConstellationLoggerHelper.exportPropertyBuilder(any(), any(), any(), any())).thenAnswer(invocation -> { 
+            Object[] args = invocation.getArguments();
+            File passedFile = (File)args[2];
+            constellationLoggerHelperStatus = (String)args[3];
+
+            // Ensure the file was not created
+            fileCreated = passedFile.exists();
+            if (fileCreated) passedFile.delete();
+
+            // Return properties (which are ignored)
+            Properties properties = new Properties();
+            return properties;
+        });
+        try {
+            SaveResultsFileWriter.writeRecordStore(plugin, tabularRecordStore);
+
+            // Execution should not make it this far, an invalid file should result in an exception.
+            Assert.assertTrue(false, "Exception is expected.");
```

Comment:
if you need the test to fail at this point, just use `fail();`

Reply:
yep, noted.